packages/design-system/src/components/NumberInput/NumberInput.tsx

```diff
@@ -78,8 +78,9 @@ const NumberInput = React.forwardRef<HTMLInputElement, NumberInputProps>(
       const parsedValue = numberParserRef.current.parse(inputValue);
 
       const newValue = isNaN(parsedValue) ? step : parsedValue + step;
+      const roundedValue = parseFloat(newValue.toFixed(10));
```

Comment:
Just wondering where the 10 value comes from?

Reply:
Yeah, I put 10 as I wasn't sure about the use cases for how many decimals I should keep, but you are right, I will derive the precision from the `step`!